File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/ClipboardBasedSnippetVariableResolver.java

package io.github.abc15018045126.sora.widget.snippet.variable;

import android.content.ClipboardManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class ClipboardBasedSnippetVariableResolver implements ISnippetVariableResolver {

    private final ClipboardManager clipboardManager;

    public ClipboardBasedSnippetVariableResolver(@Nullable ClipboardManager clipboardManager) {
        this.clipboardManager = clipboardManager;
    }

    @NonNull
    @Override
    public String[] getResolvableNames() {
        return new String[]{"CLIPBOARD"};
    }

    @NonNull
    @Override
    public String resolve(@NonNull String name) {
        if ("CLIPBOARD".equals(name)) {
            if (clipboardManager != null && clipboardManager.hasPrimaryClip()) {
                var clip = clipboardManager.getPrimaryClip();
                if (clip == null || clip.getItemCount() == 0) {
                    return "";
                }
                return clip.getItemAt(0).getText().toString();
            }
            return "";
        }
        throw new IllegalArgumentException("Unsupported variable name:" + name);
    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/ClipboardBasedSnippetVariableResolver.kt
package io.github.abc15018045126.sora.widget.snippet.variable

import android.content.ClipboardManager

class ClipboardBasedSnippetVariableResolver(private val clipboardManager: ClipboardManager?) : ISnippetVariableResolver {

    override fun getResolvableNames(): Array<String> {
        return arrayOf("CLIPBOARD")
    }

    override fun resolve(name: String): String {
        if ("CLIPBOARD" == name) {
            val clip = clipboardManager?.primaryClip
            if (clip != null && clip.itemCount > 0) {
                return clip.getItemAt(0).text?.toString() ?: ""
            }
            return ""
        }
        throw IllegalArgumentException("Unsupported variable name:$name")
    }
}
